fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java

```diff
@@ -104,13 +105,18 @@ public PhysicalPlan plan(LogicalPlan plan, PhysicalProperties outputProperties,
                 // cascades style optimize phase.
                 .setJobContext(outputProperties);
 
+        finalizeAnalyze();
         rewrite();
         optimize();
 
         // Get plan directly. Just for SSB.
         return getRoot().extractPlan();
     }
 
+    private void finalizeAnalyze() {
+        new FinalizeAnalyzeJob(plannerContext).execute();
```

Comment:
Do we really need a specific job to do this?

Reply:
yes, because we should eliminate all the logical-subquery-alias nodes, or the rewrite phase will core.